scipy/stats/_continuous_distns.py

```diff
@@ -6121,15 +6122,88 @@ def _entropy(self, c):
         data, fshape, floc, fscale = parameters
-        if floc is None:
+
+        if floc is None and fscale is None:
+            # Optimization over a system of equations is possible when the
+            # location of the data is not set. The optimization requires that
+            # the scale be free, but the shape may be fixed. The equations
+            # are from the partial derivatives of the log-likelihood function.
+
+            def get_shape(scale, location):
+                # The partial derivative with respect to the shape can be
+                # solved in closed form for the shape.
+                if fshape is None:
+                    return (data.shape[0] /
+                            np.sum(np.log((data - location) / scale)))
+                else:
+                    # if `fshape` is fixed, return that instead.
+                    return fshape
+
+            def get_location(scale):
+                # Return the numerically optimal location.
+                return np.min(data) - scale
+
+            def dLdScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return data.shape[0] * shape / scale
+
+            def dLdLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize over the scale, setting the partial derivatives
+                # w.r.t. to location and scale equal.
+                location = get_location(scale)
+                shape = get_shape(scale, location)
+                return dLdLocation(shape, location) - dLdScale(shape, scale)
+
+            def interval_contains_root(lbrack, rbrack):
+                # returns true if the signs disagree.
+                return (np.sign(fun_to_solve(lbrack)) !=
+                        np.sign(fun_to_solve(rbrack)))
+            # set brackets for `root_scalar` to use when optimizing over the
+            # scale such that a root is likely between them.
+            lbrack, rbrack = .1, np.max(data)
+            # if a root is not between the brackets, iteratively expand them
+            # until they include a sign change, checking after each bracket is
+            # modified.
+            if not interval_contains_root(lbrack, rbrack):
+                while True:
+                    lbrack /= 2
+                    if interval_contains_root(lbrack, rbrack):
+                        break
+                    rbrack *= 2
+                    if interval_contains_root(lbrack, rbrack):
+                        break
+            res = root_scalar(fun_to_solve, bracket=[lbrack, rbrack])
+
+            if res.converged:
+                fscale = res.root
+                floc = get_location(fscale)
+                fshape = get_shape(fscale, floc)
```

Comment:
```suggestion
                fshape = fshape or get_shape(fscale, floc)
```
If i'm reading correctly, this line currently overrides a `fshape` if it is provided by the user. With this change, `fshape or get_shape(fscale, floc)` will evaluate to `get_shape(fscale, floc)` if `fshape` is `None` and `fshape` otherwise.